GenericsHomework/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System.Collections;
+using System.Data;
+
+namespace GenericsHomework;
+
+public class Node<T> : ICollection<T>
+{
+    private Node<T>? _next;
+
+    public Node(T item)
+    {
+        Item = item;
+    }
+
+    private Node(T item, Node<T> next)
+    {
+        Item = item;
+        Next = next;
```

Comment:
You should just rely on your append method, what if you only want a single node in your list?
```suggestion
        Next = this;
```